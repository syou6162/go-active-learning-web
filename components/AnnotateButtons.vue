<template>
  <div>
    Annotate as: 
    <b-button size="sm" v-bind:variant="example.Label == 1 ? 'primary' : 'outline-primary'" :pressed="false" v-on:click="updateLabel(example, 1)">Positive</b-button>
    <b-button size="sm" v-bind:variant="example.Label == -1 ? 'danger' : 'outline-danger'" v-on:click="updateLabel(example, -1)">Negative</b-button>
    <b-button size="sm" v-bind:variant="example.Label == 0 ? 'secondary' : 'outline-secondary'" v-on:click="updateLabel(example, 0)">Unlabeled</b-button>
  </div>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator';
import axios from 'axios';
import Example from '~/models/Example'

@Component
export default class AnnotateButtons extends Vue {
  @Prop({required: true})
  example!: Example 
  
  updateLabel(example: Example, label: number) {
    let idToken = localStorage.getItem("CognitoIdentityServiceProvider.4ia5ifrn456rqg4vr6dqfh7e68.yasuhisa.idToken");
    let headers = { headers: { 'Authorization': idToken } };
    axios.post(
      "https://3ojd2wnlpg.execute-api.us-east-1.amazonaws.com/Prod/update_example_label", 
      {
        id: example.Id,
        label: label,
      },
      headers
    ).then(response => {
      example.Label = response.data.Label;
    }).catch(function (error) {
      alert(`${error}: Failed to annotate "${example.Title}"`);
    })
  }
}
</script>