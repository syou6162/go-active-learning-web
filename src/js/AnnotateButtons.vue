<template>
  <div>
    Annotate as: 
    <b-button size="sm" v-bind:variant="[example.Label == 1 ? 'primary' : 'outline-primary']" :pressed="false" v-on:click="updateLabel(example, 1)">Positive</b-button>
    <b-button size="sm" v-bind:variant="[example.Label == -1 ? 'danger' : 'outline-danger']" v-on:click="updateLabel(example, -1)">Negative</b-button>
    <b-button size="sm" v-bind:variant="[example.Label == 0 ? 'secondary' : 'outline-secondary']" v-on:click="updateLabel(example, 0)">Unlabeled</b-button>
  </div>
</template>

<script>
import axios from 'axios';

export default {
  props: ['example'],
  methods: {
    updateLabel(example, label) {
      axios.post(
        "https://3ojd2wnlpg.execute-api.us-east-1.amazonaws.com/Prod/update_example_label", 
        {
          url: example.Url,
          label: label,
        },
      ).then(response => {
        example.Label = label;
      }).catch(function (error) {
        alert(`Failed to annotate "${example.Title}"`);
      })
    },
  }
}
</script>
