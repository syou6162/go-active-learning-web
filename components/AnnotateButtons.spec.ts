import { createLocalVue, mount } from '@vue/test-utils'
import { ButtonPlugin } from 'bootstrap-vue'
import AnnotateButtons from './AnnotateButtons.vue'

const localVue = createLocalVue();
localVue.use(ButtonPlugin)

describe('AnnotateButtons', () => {
  test('ラベルが0のときはunlabeldが有効になっている', () => {
    const wrapper = mount(AnnotateButtons, {
      localVue,
      propsData: {
        example: {
          Id: 11111,
          Label: 0
        }
      },
    })
    expect(wrapper.exists()).toBeTruthy()
    expect(wrapper.findAll('button').length).toBe(3)

    expect(wrapper.findAll('button').at(0).classes()).toContain('btn-outline-primary')
    expect(wrapper.findAll('button').at(1).classes()).toContain('btn-outline-danger')
    expect(wrapper.findAll('button').at(2).classes()).toContain('btn-secondary')

    expect(wrapper.element).toMatchSnapshot()
  })
})
