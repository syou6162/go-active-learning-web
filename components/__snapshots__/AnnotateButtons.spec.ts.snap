// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnnotateButtons ラベルが0のときはunlabeldが有効になっている 1`] = `
<div>
  
  Annotate as: 
  
  <button
    aria-pressed="false"
    autocomplete="off"
    class="btn btn-outline-primary btn-sm"
    type="button"
  >
    
    Positive
  
  </button>
   
  <button
    class="btn btn-outline-danger btn-sm"
    type="button"
  >
    
    Negative
  
  </button>
   
  <button
    class="btn btn-secondary btn-sm"
    type="button"
  >
    
    Unlabeled
  
  </button>
</div>
`;
